nautobot/extras/management/__init__.py

```diff
@@ -184,98 +269,113 @@ def create_custom_statuses(
+                        f"Unexpected error while running data migration to populate {metadata_model} for {model_path}: {err}"
                     ) from err
 
             # Make sure the content-type is associated.
             if content_type not in obj.content_types.all():
                 obj.content_types.add(content_type)
 
             if created and verbosity >= 2:
-                print(f"      Adding and linking status {obj.name}", flush=True)
+                print(f"      Adding and linking {metadata_model} {obj.name}", flush=True)
                 added_total += 1
             elif not created and verbosity >= 2:
-                print(f"      Linking to existing status {obj.name}", flush=True)
+                print(f"      Linking to existing {metadata_model} {obj.name}", flush=True)
                 linked_total += 1
 
     if verbosity >= 2:
-        print(f"    Added {added_total}, linked {linked_total} status records")
+        print(f"    Added {added_total}, linked {linked_total} {metadata_model} records")
 
 
-def clear_status_choices(
+def _clear_custom_role_or_status_instances(
     apps=global_apps,
     schema_editor=None,
     verbosity=2,
     models=None,
+    metadata_model=None,
     clear_all_model_statuses=True,
     **kwargs,
 ):
     """
-    Remove content types from statuses, and if no content types remain, delete the statuses as well.
+    Remove content types from statuses/roles, and if no content types remain, delete the status/role instance as well.
+    Args:
+        models (dict): A list of model contenttype strings e.g. models=["circuits.Circuit", "dcim.Cable", "dcim.Device","dcim.PowerFeed"]
+        metadata_model (str): "role" or "status"
+        clear_all_model_statuses (bool): Set it to True will clear all statuses for this model. Set it to False will only clear default statuses for this model.
     """
     if "test" in sys.argv:
         # Do not print output during unit testing migrations
         verbosity = 1
 
-    if verbosity >= 0:
-        print("\n", end="")
-
-    if not models:
-        models = CHOICESET_MAP.keys()
-
-    Status = apps.get_model("extras.Status")
-    ContentType = apps.get_model("contenttypes.ContentType")
+    choiceset_map = {}
+    if metadata_model.lower() == "role":
+        choiceset_map = ROLE_CHOICESET_MAP
+        if not models:
+            models = choiceset_map.keys()
+    elif metadata_model.lower() == "status":
+        choiceset_map = STATUS_CHOICESET_MAP
+        if not models:
+            models = choiceset_map.keys()
+    else:
+        raise ValueError("only status and role are supported for _clear_custom_role_or_status_instances")
+
+    # Prep the app and get the model dynamically
+    try:
+        Metadata_Model = apps.get_model(f"extras.{metadata_model}")
+        ContentType = apps.get_model("contenttypes.ContentType")
+    except LookupError:
+        return
```

Comment:
What's the case where we'd get this exception? Should we log anything and/or re-raise the exception instead of just silently returning?

Same question on the forward function above, actually.

Reply:
I think the only case this would raise a lookup error is that the ContentType and Status/Role migration has not been run yet when the user is using this method. I will output a log message in this case.